packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1881,19 +1881,19 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     newRoute.install(_currentOverlayEntry);
     _history[index] = newRoute;
     newRoute.didPush().whenCompleteOrCancel(() {
-      // The old route's exit is not animated. We're assuming that the
```

Comment:
This comment is still true. When the route is removed, that removal is not animated.

Reply:
Maybe move the comment inside the if, though.